File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int main(){
    cin.tie(0);cout.tie(0);ios::sync_with_stdio(false);
    int n, m; cin >> n >> m;
    vector<vector<int>> edge(n+1);
    vector<bool> visited(n+1, false);
    for(int i = 0; i < m; i++){
        int x, y; cin >> x >> y;
        edge[x].push_back(y);
        edge[y].push_back(x);
    }
    int ans = 0;
    for(int i = 1; i <= n; i++){
        if(!visited[i]){
            visited[i] = true;
            stack<int> st;
            st.push(i);
            ans++;
            while(st.size()){
                int cur = st.top(); st.pop();
                for(auto nxt : edge[cur]){
                    if(!visited[nxt]){
                        visited[nxt] = true;
                        st.push(nxt);
                    }
                }
            }
        }
    }
    cout << ans;
    return 0;
}

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.py
import sys
print = lambda *args, sep=" ", end="\n": sys.stdout.write(sep.join(map(str, args)) + end)
input = lambda: sys.stdin.readline().rstrip('\r\n')

def main():
    n, m = map(int, input().split())
    edge = [[] for _ in range(n+1)]
    visited = [False]*(n+1)
    for _ in range(m):
        x, y = map(int, input().split())
        edge[x].append(y)
        edge[y].append(x)
    ans = 0
    for i in range(1, n+1):
        if not visited[i]:
            visited[i] = True
            st = [i]
            ans += 1
            while st:
                cur = st.pop()
                for nxt in edge[cur]:
                    if not visited[nxt]:
                        visited[nxt] = True
                        st.append(nxt)
    print(ans)
    return 0
if __name__ == '__main__':
    sys.exit(main())